src/js/view/utils/resize-listener.js

```diff
@@ -50,44 +67,32 @@ export default class ResizeListener {
         this.expandElement = expandElement;
         this.expandChild = expandChild;
         this.contractElement = contractElement;
-        this.hiddenElement = element.appendChild(resizeElement);
+        this.hiddenElement = resizeElement;
         this.element = element;
-        this.callback = callback;
-        this.resizeRaf = -1;
-        this.lastWidth = this.currentWidth = 0;
-        this.scrollListener = (e) => {
-            cancelAnimationFrame(this.resizeRaf);
-            this.resizeRaf = requestAnimationFrame(() => {
-                const currentWidth = this.currentWidth = element.offsetWidth;
-                if (this.lastWidth === currentWidth) {
-                    return;
-                }
-                this.callback(e, currentWidth);
-            });
-            this.resetTriggers();
-        };
-
-        requestAnimationFrame(() => {
-            this.currentWidth = element.offsetWidth;
-            this.resetTriggers();
-            element.addEventListener('scroll', this.scrollListener, true);
-        });
-    }
-
-    resetTriggers() {
-        const currentWidth = this.currentWidth;
-        this.contractElement.scrollLeft = currentWidth * 2;
-        style(this.expandChild, { width: currentWidth + 1 });
-        this.expandElement.scrollLeft = currentWidth + 1;
-        this.lastWidth = currentWidth;
+        this.view = view;
+        this.model = model;
+        this.width = 0;
+        this.resized = false;
+        if (element.firstChild) {
+            element.insertBefore(resizeElement, element.firstChild);
+        } else {
+            element.appendChild(resizeElement);
+        }
+        element.addEventListener('scroll', scrollListener, true);
+        instances.push(this);
+        scrollListener();
     }
 
     destroy() {
-        if (this.callback) {
-            this.element.removeEventListener('scroll', this.scrollListener, true);
+        if (this.view) {
```

Comment:
wouldn't we want to splice this out regardless or whether or not there's a view? not too clear on why the destroy is dependent on this view check when we don't have a similar check in the constructor where this stuff is set up.the event listener could dangle here or anything else. i might be overlooking something as well!

Reply:
`view` is assigned a value in the constructor, and set to `null` when destroyed. The check is so we don't run any of the destroy logic twice. It's the `removeChild` call that would fail if called twice, but there's not reason to run any of the other code if `view` or `model` was nulled.